modules/UI/videolayout/ConnectionIndicator.js

```diff
@@ -351,23 +357,30 @@ ConnectionIndicator.prototype.remove = function() {
 };
 
 /**
- * Updates the UI which displays warning about user's connectivity problems.
+ * Updates the UI which displays or not a warning about user's connectivity
+ * problems.
  *
- * @param {boolean} isActive true if the connection is working fine or false if
- * the user is having connectivity issues.
+ * @param {ParticipantConnectionStatus} connectionStatus
  */
 ConnectionIndicator.prototype.updateConnectionStatusIndicator
-    = function (isActive) {
-        this.isConnectionActive = isActive;
-        if (this.isConnectionActive) {
-            $(this.interruptedIndicator).hide();
-            $(this.emptyIcon).show();
-            $(this.fullIcon).show();
-        } else {
-            $(this.interruptedIndicator).show();
-            $(this.emptyIcon).hide();
-            $(this.fullIcon).hide();
-        }
+= function (connectionStatus) {
+    this.connectionStatus = connectionStatus;
+    if (connectionStatus === ParticipantConnectionStatus.INTERRUPTED) {
+        $(this.interruptedIndicator).show();
+        $(this.emptyIcon).hide();
+        $(this.fullIcon).hide();
+        $(this.ninjaIndicator).hide();
+    } else if (connectionStatus === ParticipantConnectionStatus.INACTIVE) {
+        $(this.interruptedIndicator).hide();
+        $(this.emptyIcon).hide();
+        $(this.fullIcon).hide();
+        $(this.ninjaIndicator).show();
+    } else {
+        $(this.interruptedIndicator).hide();
+        $(this.emptyIcon).show();
+        $(this.fullIcon).show();
+        $(this.ninjaIndicator).hide();
+    }
```

Comment:
Well if it's harder for you to figure out on a glance then that's reason enough for me not to recommend it anymore. What's your thought on storing the instance variables as jquery objects? (Don't actually do it though because I'll probably rewrite ConnectionIndicator in the coming weeks anyway.)